pkg/config/validator.go

```diff
@@ -364,6 +409,78 @@ func (p *validator) validateAspectRules(rules []*pb.AspectRule, path string, val
 	return numAspects, ce
 }
 
+func (p *validator) validateRules(rules []*pb.Rule, path string) (ce *adapter.ConfigErrors) {
```

Comment:
This is almost identical to `validateAspectRules`, can they be combined in any meaningful way? At least some of the logic for walking the config block should be shareable. We might want to swap over to using the visitor pattern to walk the config and dispatch to the various validators.